src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GoogleServicesDAO.scala

```diff
@@ -12,16 +12,14 @@ import scala.concurrent.Future
 abstract class GoogleServicesDAO(groupsPrefix: String) {
 
   // returns a workspaceID
-  def setupWorkspace(userInfo: UserInfo, projectId: String, workspaceId: String, workspaceName: WorkspaceName): Future[Unit]
+  def setupWorkspace(userInfo: UserInfo, projectId: String, workspaceId: String, workspaceName: WorkspaceName): Future[GoogleWorkspaceInfo]
 
   def createCromwellAuthBucket(billingProject: RawlsBillingProjectName): Future[String]
 
-  def deleteWorkspace(bucketName: String, monitorRef: ActorRef): Future[Any]
+  def deleteWorkspace(bucketName: String, accessGroups: Seq[RawlsGroup], monitorRef: ActorRef): Future[Any]
 
   def deleteBucket(bucketName: String, monitorRef: ActorRef): Future[Any]
 
-  def getBucketName(workspaceId: String) = s"${groupsPrefix}-${workspaceId}"
```

Comment:
name of the bucket is stored on the workspace, constructing the name from the workspaceId everywhere is incorrect